src/engine/Eval_generic.ml

```diff
@@ -139,6 +141,18 @@ let value_of_lit ~code x =
   | G.Float (Some f, _t) -> Float f
   | _ -> raise (NotHandled code)
 
+(* helper function for comparing dates *)
+let date_compare date1 date2 =
+  match (date1, date2) with
+  | (y1, m1, d1), (y2, m2, d2) ->
+      if y1 < y2 then -1
```

Comment:
I believe there's `Int.compare : int -> int -> int`, so you could do something like
```
let y_cmp, m_cmp, d_cmp = Int.compare (y1, y2), Int.compare (m1, m2), Int.compare (d1, d2) in
if y_cmp = 0 then
  ...
else ...
```
(less efficient, but negligibly so)

It would be nice if we had a first-class way of doing this logic, like some kind of comparison chaining operator, in our common library, but AFAIK we don't. Oh well.

Reply:
Wait! I'm wrong! There's `Stdlib.compare`, so we could `Stdlib.compare` on some 3-place tuples.

Now, I hate polymorphic compare as much as the next person, but this code isn't prone to refactoring, so we should be OK.

Reply:
I have an experiment to derive compare (https://github.com/returntocorp/semgrep/pull/7996)